code/modules/hydroponics/grown.dm

```diff
@@ -160,11 +160,19 @@
 		reagents.del_reagent(/datum/reagent/consumable/nutriment)
 		reagents.del_reagent(/datum/reagent/consumable/nutriment/vitamin)
 
-// For item-containing growns such as eggy or gatfruit
+/*
+ * Attack self for growns
+ *
+ * Puts trash item into the hand of user attack selfing, or drops it on the ground
+ *
+ * Then deletes the grown object
+ */
 /obj/item/reagent_containers/food/snacks/grown/shell/attack_self(mob/user)
 	var/obj/item/T
 	if(trash)
-		T = generate_trash()
+		T = generate_trash(src)
+		//put in hands or drop to ground
+		user.put_in_hands(T, user.active_hand_index, TRUE)
+		//Delete *AFTER* we have moved the trash item out of our contents
 		qdel(src)

```

Comment:
Hmm I just realised this will be taking up the hand slot, so we really need to spawn the item on the floor/turf, then do this, then try to move the item into the users hands